src/main/java/emissary/util/ImageIOClassLoader.java

```diff
@@ -0,0 +1,29 @@
+package emissary.util;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+import emissary.config.ConfigEntry;
+
+/**
+ * Load the ImageIOClass implementation from the file system using the emissary.config.Configurator interface.
+ */
+public class ImageIOClassLoader {
+
+    /**
+     * Read the ImageIO and Surplus ImageIO classes that will be registered/ deregistered from the config file to initialize
+     * the ImageIOClass class
+     */
+    public static void initialize() {
+        if (ImageIOClass.isInitialized()) {
```

Comment:
There appears to be a time-of-check / time-of-access race condition in this method.  The `ImageIOClass.isInitialized()` and `ImageIOClass.initialize(...)` methods each use the `synchronized` keyword to ensure that single-threaded access to those methods indepndently, but there's no synchronization across those.  That means the body of `ImageIOClass.initialize(...)` could execute multiple times.   Should be able to mitigate by 

1. locking on a object within `ImageIOClass`, 
2. adding another check at the start of the `ImageIO.initialize` method, or 
3. making `ImageIOClassLoader.initialize()` synchronized.

Reply:
Made ImageIOClassLoader.initialize() synchronized.